common4j/src/main/com/microsoft/identity/common/java/logging/Logger.java

```diff
@@ -457,5 +446,26 @@ private static String getUTCDateTimeAsString() {
 
         return dateFormat.format(new Date());
     }
+
+    /**
+     * Get only the required metadata from the DiagnosticContext to plug it in the log lines.
+     * Here thread_name is taken from DiagnosticContext, and correlationId from input parameter.
+     * The need for this is because DiagnosticContext contains additional metadata which is not always required to be logged.
+     *
+     * @return String The concatenation of thread_name and correlation_id to serve as the required metadata in the log lines.
+     */
+    private static synchronized String getDiagnosticContextMetadata(String correlationId) {
```

Comment:
suggest marking correlationId nullable, and we can use this method everywhere. (we don't need the other one).

Reply:
From code it seems, a mainstream usecase is to not pass correlationId, and expect it to be taken from DiagnosticContext's requestContext. If we keep this method, many callers will simply make getDiagnosticContextMetadata(null) call by design. So, I felt it's better to keep two separate methods. Thoughts? 
getDiagnosticContextMetadata() is public (from both common and common4j). Is that supposed to be part of public API of this lib? If so probably both versions of this method need to be public (or just one method when correlationId is marked as nullable).